core/src/main/scala/org/apache/spark/status/api/v1/SimpleDateParam.scala

```diff
@@ -25,8 +25,19 @@ import javax.ws.rs.core.Response.Status
 import scala.util.Try
 
 private[v1] class SimpleDateParam(val originalValue: String) {
+
+  private val formats: Seq[SimpleDateFormat] = {
```

Comment:
Now this is a field though. There's no need for that. The format objects can be created on demand to parse the input. If this `SimpleDateParam` object is short-lived I suppose it does not matter much but I would have just inlined all this.